### PR TITLE
MGMT-16332: enable LVM and CNV multinode for 4.15

### PR DESCRIPTION
**Pull request by eifrach**
## List all the issues related to this PR

- [ ] New Feature 
- [ ] Enhancement 
- [ ] Bug fix
- [ ] Tests
- [ ] Documentation
- [ ] CI/CD 

## What environments does this code impact?

- [ ] Automation (CI, tools, etc)
- [ ] Cloud
- [ ] Operator Managed Deployments
- [x] None

## How was this code tested?

- [ ] assisted-test-infra environment
- [ ] dev-scripts environment
- [ ] Reviewer's test appreciated
- [ ] Waiting for CI to do a full test run
- [ ] Manual (Elaborate on how it was tested)
- [x] No tests needed

## Checklist

- [ ] Title and description added to both, commit and PR.
- [ ] Relevant issues have been associated (see [CONTRIBUTING] guide)
- [ ] This change does not require a documentation update (docstring, `docs`, README, etc)
- [ ] Does this change include unit-tests (note that code changes require unit-tests)

## Reviewers Checklist

- Are the title and description (in both PR and commit) meaningful and clear?
- Is there a bug required (and linked) for this change?
- Should this PR be backported?

[Kubernetes community documentation]: https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md#commit-message-guidelines
[CONTRIBUTING]: https://github.com/openshift/assisted-service/blob/master/CONTRIBUTING.md

### CODE REVIEW

File: internal/operators/manager.go [hunk 1]
```diff
@@ -463,13 +449,24 @@ func EnsureLVMAndCNVDoNotClash(cluster *common.Cluster, openshiftVersion string,
...
+    return nil
+	}
+
+	// Openshift version greater or Equal to 4.12.0 support cnv and lvms
+  if isGreaterOrEqual, _ := common.BaseVersionGreaterOrEqual(lvm.LvmsMinOpenshiftVersion4_12, openshiftVersion); !isGreaterOrEqual {
+		return errors.Errorf("Currently, you can not install OpenShift Data Foundation Logical Volume Manager operator at the same time as Virtualization operator.") 
 	}
 
-	if lvmEnabled && !common.IsSingleNodeCluster(cluster) {
+  if isGreaterOrEqual, _ := common.BaseVersionGreaterOrEqual(lvm.LvmMinMultiNodeSupportVersion, openshiftVersion); isGreaterOrEqual {
```
Comment (gamli75) [inline internal/operators/manager.go:461]: I think that this condition should be before 
// Openshift version greater or Equal to 4.12.0 support cnv and lvms

File: internal/operators/manager.go [hunk 2]
```diff
@@ -463,13 +449,24 @@ func EnsureLVMAndCNVDoNotClash(cluster *common.Cluster, openshiftVersion string,
...
+    return nil
+  }
+
+  // before 4.15 muli node LVM not supported
```
Comment (gamli75) [inline internal/operators/manager.go:465]:
```suggestion
  // before 4.15 multi node LVM not supported
```

File: internal/operators/manager.go [hunk 3]
```diff
@@ -463,13 +449,24 @@ func EnsureLVMAndCNVDoNotClash(cluster *common.Cluster, openshiftVersion string,
...
+	if !common.IsSingleNodeCluster(cluster) {
```
Comment (gamli75) [inline internal/operators/manager.go:466]: the error message "OpenShift Data Foundation Logical Volume Manager operator is only supported for Single Node Openshift" is not correct

File: internal/operators/manager.go (hunk 1 above)
Comment (eifrach) [reply internal/operators/manager.go]: yes I check for single mode after this if, on 4.15 we don't need to check if it `singleNodeCluster`

File: internal/operators/manager.go (hunk 3 above)
Comment (eifrach) [reply internal/operators/manager.go:465]: That was the message before than, What should it be?

Comment (gamli75) [reply internal/operators/manager.go:465]: should be the same message as:
message := fmt.Sprintf("Logical Volume Manager is only supported for highly available openshift with version %s or above", LvmMinMultiNodeSupportVersion) but I don't like the highly available openshift

Comment (gamli75) [reply internal/operators/manager.go:465]: @nirfarkas needs your help with the error message for LVMS on multi node cluster in OCP version < 4.15

File: internal/operators/manager.go (hunk 2 above)
Comment (eifrach) [reply internal/operators/manager.go]: fixed

**Comment by eifrach** [comment]
will be created in a new PR
/close

**Pull request closed without merging**